File: Day-32/ArrayIntersection.py
def ArrayIntersection(arr1, arr2):
    i = 0
    j = 0
    returnArray =[]
    while(i<len(arr1) and j < len(arr2)):
        if(arr1[i]==arr2[j]):
            returnArray.append(arr2[j])
            i += 1
            j += 1
        elif(arr1[i] < arr2[j]):
            i += 1
        else:
            j += 1

    return returnArray

print(ArrayIntersection([1,4,5], [3,4,5]))

File: Day-32/ArrayIntersection.js
function ArrayIntersection(array1, array2) {
  let returnArray = [];
  let i = 0;
  let j = 0;
  while (i < array1.length && j < array2.length) {
    if (array1[i] === array2[j]) {
      returnArray.push(array1[j]);
      i++;
      j++;
    } else if (array1[i] < array2[j]) {
      i++;
    } else {
      j++;
    }
  }

  return returnArray;
}

console.log(ArrayIntersection([3], [6]));
